File: articles/decision-science-judgment-under-uncertainty/julia/expected_value.jl
# Decision Science: Expected Value in Julia
# Educational example only.

alternatives = Dict(
    "Incremental Program Upgrade" => [(0.65, 72.0), (0.35, 38.0)],
    "Targeted Resilience Investment" => [(0.55, 84.0), (0.45, 52.0)],
    "Large-Scale Transformation" => [(0.45, 96.0), (0.55, 30.0)]
)

function expected_value(outcomes)
    return sum(probability * value for (probability, value) in outcomes)
end

for (alternative, outcomes) in alternatives
    println(alternative, ": ", expected_value(outcomes))
end

File: articles/decision-science-judgment-under-uncertainty/fortran/expected_value.f90
program expected_value_model
  implicit none

  real, dimension(2) :: probabilities
  real, dimension(2) :: values
  real :: ev

  probabilities = (/0.65, 0.35/)
  values = (/72.0, 38.0/)

  ev = sum(probabilities * values)

  print *, "Expected value:", ev

end program expected_value_model
